test/functional/test_framework/util.py

```diff
@@ -296,16 +296,23 @@ def initialize_datadir(dirname, n, chain):
     datadir = get_datadir_path(dirname, n)
     if not os.path.isdir(datadir):
         os.makedirs(datadir)
-    # Translate chain name to config name
+    write_config(os.path.join(datadir, "bitcoin.conf"), n, chain)
+    os.makedirs(os.path.join(datadir, 'stderr'), exist_ok=True)
+    os.makedirs(os.path.join(datadir, 'stdout'), exist_ok=True)
+    return datadir
+
+
+def write_config(config_path, n, chain, extra_config=""):
+    # Translate chain subdirectory name to config name
     if chain == 'testnet3':
-        chain_name_conf_arg = 'testnet'
-        chain_name_conf_section = 'test'
+        chain_name = 'test'
+    elif not chain:
+        chain_name = 'main'
     else:
-        chain_name_conf_arg = chain
-        chain_name_conf_section = chain
-    with open(os.path.join(datadir, "bitcoin.conf"), 'w', encoding='utf8') as f:
-        f.write("{}=1\n".format(chain_name_conf_arg))
-        f.write("[{}]\n".format(chain_name_conf_section))
+        chain_name = chain
+    with open(config_path, 'w', encoding='utf8') as f:
+        f.write("chain={}\n".format(chain_name))
+        f.write("[{}]\n".format(chain_name))
```

Comment:
`feature_config_args` needs to be updated for this change as well

Reply:
> `feature_config_args` needs to be updated for this change as well

Why/how?

Reply:
It parses the debug.log to check for the chain section or something

Reply:
(See the test failure)

Reply:
Oh, this seems to be a silent merge conflict with #16115. Doesn't happen on my local branch since I never rebased it. Will debug